File: cours-filters-java/src/main/java/fr/isima/filters/CounterFilter.java
package fr.isima.filters;

import javax.servlet.*;
import javax.servlet.annotation.WebFilter;
import java.io.IOException;
import java.util.concurrent.atomic.AtomicLong;

/**
 * Filtre permettant de compter le nombre de requêtes.
 * <p>
 * A noter que l'utilisation de {@link System#out} est généralement interdite
 * dans les projets en production, et que les informations de ce type sont
 * écrites par exemple dans des fichiers de logs.
 * </p>
 *
 * @author Benjamin
 */
@WebFilter("/*")
public class CounterFilter implements Filter {

    private AtomicLong counter = new AtomicLong();

    @Override
    public void doFilter(ServletRequest req, ServletResponse rep,
                         FilterChain chain) throws IOException, ServletException {

        System.out.println("Nombre de requetes " + counter.incrementAndGet());
        chain.doFilter(req, rep);
    }

    @Override
    public void init(FilterConfig arg0) throws ServletException {
    }

    @Override
    public void destroy() {
    }


    long getCounter() {
        return counter.get();
    }
}

File: cours-filters-kotlin/src/main/kotlin/fr/isima/filters/CounterFilter.kt
package fr.isima.filters

import java.util.concurrent.atomic.AtomicLong
import javax.servlet.*
import javax.servlet.annotation.WebFilter

/**
 * Filtre permettant de compter le nombre de requêtes.
 *
 *
 * A noter que l'utilisation de [System.out] est généralement interdite
 * dans les projets en production, et que les informations de ce type sont
 * écrites par exemple dans des fichiers de logs.
 *
 *
 * @author Benjamin
 */
@WebFilter("/*")
class CounterFilter(private val counter: AtomicLong = AtomicLong(0)) : Filter {

    override fun doFilter(req: ServletRequest, rep: ServletResponse, chain: FilterChain) {
        println("Nombre de requetes " + counter.incrementAndGet())
        chain.doFilter(req, rep)
    }

    internal fun getCounter() = counter.get()

    override fun init(arg0: FilterConfig) {}
    override fun destroy() {}
}
